spring-integration-core/src/main/java/org/springframework/integration/config/AbstractSimpleMessageHandlerFactoryBean.java

```diff
@@ -312,7 +317,7 @@ public boolean isSingleton() {
 		return true;
 	}
 
-	private static Object extractTarget(Object object) {
+	private static @Nullable Object extractTarget(@Nullable Object object) {
```

Comment:
I think this method has to be changed to be non-nullable.
And we should have something like in the body:
```
return Objects.requireNonNullElse(AopProxyUtils.getSingletonTarget(object), object);
```

I doubt that we need recursion here.
But I might be wrong if some tests would fail.
Forth checking anyway!